tardis/adapters/sites/kubernetes.py

```diff
@@ -0,0 +1,176 @@
+                match_labels={"app": self.machine_type_configuration.label}
+            ),
+            template=client.V1PodTemplateSpec(),
+        )
+
+        container = client.V1Container(
+            image=self.machine_type_configuration.image,
+            args=self.machine_type_configuration.args.split(","),
+            name=resource_attributes.drone_uuid,
+            resources=client.V1ResourceRequirements(
+                requests={
+                    "cpu": self.machine_meta_data.Cores,
+                    "memory": self.machine_meta_data.Memory,
+                }
+            ),
+        )
+
+        spec.template.metadata = client.V1ObjectMeta(
+            name=self.machine_type_configuration.label,
+            labels={"app": self.machine_type_configuration.label},
+        )
+
+        spec.template.spec = client.V1PodSpec(containers=[container])
+        body = client.V1Deployment(
+            metadata=client.V1ObjectMeta(name=resource_attributes.drone_uuid),
+            spec=spec,
+        )
+
+        responseTemp = await self.client.create_namespaced_deployment(
+            namespace="default", body=body
+        )
+
+        response = dict(
+            {
+                "uid": responseTemp.metadata.uid,
+                "name": responseTemp.metadata.name,
+                "type": "Progressing",
+            }
+        )
+
+        return self.handle_response(response)
+
+    async def resource_status(
+        self, resource_attributes: AttributeDict
+    ) -> AttributeDict:
+        list = await self.client.list_namespaced_deployment(namespace="default")
+
+        response = dict(
+            {
+                "uid": resource_attributes.remote_resource_uuid,
+                "name": resource_attributes.drone_uuid,
+                "type": "Deleted",
+            }
+        )
+
+        for x in list.items:
+            if x.metadata.name == resource_attributes.drone_uuid:
+                responseTemp = await self.client.read_namespaced_deployment(
+                    name=resource_attributes.drone_uuid, namespace="default"
+                )
+
+                if responseTemp.spec.replicas == 0:
+                    response = dict(
+                        {
+                            "uid": responseTemp.metadata.uid,
+                            "name": responseTemp.metadata.name,
+                            "type": "Stopped",
+                        }
+                    )
+                else:
+                    response = dict(
+                        {
+                            "uid": responseTemp.metadata.uid,
+                            "name": responseTemp.metadata.name,
+                            "type": responseTemp.status.conditions[0].type,
+                        }
+                    )
+
+        return self.handle_response(response)
+
+    async def stop_resource(self, resource_attributes: AttributeDict):
+        body = await self.client.read_namespaced_deployment(
```

Comment:
In relation to ``resource_status`` above: Is there a reason why one can directly access the *specific* deployment here, but needs to list-and-search all deployments above?

Reply:
Since a deployment doesn't exist after it is deleted , i checked if the deployment exists. Changed the Loop to try: except: .

Reply:
Neater and (hopefully ^^) faster. Nice!